### DISCUSSION TITLE
How can i get progress record and limit size of file record ?

### DISCUSSION THREAD

**Discussion by DangLjnh** [katspaugh/wavesurfer.js#3378] [Q&A]
I'm using the RecordPlugin to capture audio. How can I check the recording progress, specifically the elapsed recording time? Additionally, what steps should I take to set a size limit for the recorded file? Thank you for your assistance.

**Comment by katspaugh** [comment]
You can track time by setting a variable to `Date.now()` when recordinds starts, and then subtracting it from the current time like this:

```js
const startTime = Date.now()

const getRecordingTime = () => {
  return Date.now() - startTime
}

startRecording()
```

**Reply by DangLjnh** [reply]
Thanks a lot, i got it!
